server/model/monitor.js

```diff
@@ -28,9 +29,12 @@ class Monitor extends BeanModel {
      * Only show necessary data to public
      */
     async toPublicJSON() {
+        const maintenance = await R.getAll("SELECT mm.*, maintenance.start_date, maintenance.end_date FROM monitor_maintenance mm JOIN maintenance ON mm.maintenance_id = maintenance.id WHERE mm.monitor_id = ? AND datetime(maintenance.start_date) <= datetime('now', 'localtime') AND datetime(maintenance.end_date) >= datetime('now', 'localtime')", [this.id]);
```

Comment:
We have 3 this same queries in this file, 4th this same query is in `api-router.js`

Reply:
I understand, can I ask your opinion?

The states of individual monitors are currently evaluated according to the state of the last heartbeat.
If I used this system in case of maintenance, any change in the maintenance settings (adding / removing the monitor from maintenance, changing the maintenance time) would not affect the monitor and the status page until the next heartbeat of the monitor.
By directly checking whether or not the monitor is in maintenance, it is possible to mark the monitor as in maintenance before a heartbeat occurs.
For this reason, this SQL query is also called when requesting monitor information instead of being called only twice (on the status page and at heartbeat).

In my opinion, this is more user-friendly, because the fact that the monitor is in maintenance or not can be seen immediately without waiting, even on the status page.

Do you think it will be better to leave the status display according to the last heartbeat, or do you find the my current solution better?

Thanks for the feedback

Reply:
Thank you for your feedback.

In this case, 2 out of 4 calls are absolutely necessary (calls from the status page - display of maintenance information on the status page and at each beat - finding out if the monitor is under maintenance) and if I am not mistaken, in the methods "toJSON()" and  "toPublicJSON()" this calls are also required to let the current solution described above work.

If I'm wrong, feel free to tell me your opinion. 

Thank you

Reply:
I don't know . I just wanted to have this long query-code line in one place (helper function etc)

Reply:
Oh, understand. It shouldn't be problem I think, I think I'll be able to move it into one helper function and just call it.

Few minutes later I commited some changes, so I'll check / do this tomorrow I guess.

Reply:
Done. It's now in its own method.